File: swordOffer/replaceSpace/replaceSpace.cc
#include <iostream>
#include <string>

using namespace std;

class Solution {
 public:
  string replaceSpace(string s) {

    string res;
    for (char& c: s) {
      if (c == ' ') {
        res.push_back('%');
        res.push_back('2');
        res.push_back('0');
      } else {
        res.push_back(c);
      }
    }

    return res;
  }
};

File: swordOffer/go/arrStr/replaceSpace.go
package main

import "strings"

/*
func replaceSpace(s string) string {
	if s == "" {
		return ""
	}

	var res string
	for i := range s {
		if s[i] == ' ' {
			res += "%20"
		} else {
			res += string([]byte{s[i]})
		}
	}

	return res
}
*/

func replaceSpace(s string) string {

	var b strings.Builder
	for i := range s {
		if s[i] == ' ' {
			b.WriteString("%20")
		} else {
			b.WriteByte(s[i])
		}
	}

	return b.String()
}
